pkg/features/embedded/boot-assessment/system/oem/08_boot_assessment.yaml

```diff
@@ -0,0 +1,74 @@
+# Boot assessment works this way:
+# - After upgrade we set the boot_assessment_check=yes variable, this way grub will
+#   include `elemental.health_check` flag in kernel command line.
+# - On reboot grub detects the `boot_assessment_check=yes` and sets `last_boot_attempt`
+#   to store the menu entry that is currently attempting to boot.
+# - On boot failure a reboot is triggered and grub will compute the next boot option from
+#   `last_boot_attempt` variable set in previous boot, update this variable and try again.
+# - If boot succeeds the elemental-boot-assessment.service will always clear
+#   `last_boot_attempt` and clear `boot_assessment_check` only if it booted from the
+#   active system.
+
+name: "Boot assessment"
+stages:
+    after-install-chroot:
+    # The boot assessment logic is added here as part of a bootassessment.cfg file. This file is
+    # sourced in /grubcustom/custom.cfg file, which is only meant to source additional files.
+    # So far there is no regexp module in most grub2 efi signed images, hence it is not possible to
+    # to iterate over the configuration files in /grubcustom folder to load them all. Instead the
+    # pattern is to manually append additional sourced files in /grubcustom/custom.cfg.
```

Comment:
I could not manage to find a proper way of doing it. I did not want to include this in grub.cfg because this is coupled with the boot-assessment feature and grub is provided by another independent feature. I like the idea of keeping the option of installing without any kind of boot-assessment logic.

My desired setup was defining a `/grubcustom` folder for which the main `grub.cfg` would source all the existing configuration files, this way we could easily include additional custom changes besides the `bootassessment.cfg`. However I could not find a way to iterate over files found in a path, this could be possible by using the `regexp` grub2 module, but this does not come included signed efi images by default.